### ISSUE TITLE
Drawing issues with ranger

### ISSUE THREAD

**Issue by dgengtek** [tmux/tmux#1469]
This is probably an issue with my environment setup. I am not sure if I should report this here or on the ranger project since it generally works inside tmux but only when starting directly via a new window it fails.

Running `$ tmux neww ranger` does the following drawing 
![ranger_tmux_issue](https://user-images.githubusercontent.com/13528886/45252466-0d367700-b357-11e8-9fce-aa180837c487.png)
which draws correctly when I just run `$ ranger` inside an already opened tmux window.

I though that it would be an issue with my TERM. Running `echo $TERM` in the bash shell gets me 'screen' while `$ tmux showenv` TERM gives me `TERM=rxvt-unicode-256color`.

I do not set TERM via bash. This is done by my terminal _rxvt-unicode (urxvt) v9.22 - released: 2016-01-2_ which is `TERM=rxvt-unicode-256color` when starting the terminal without _tmux 2.7_.

The only relevant tmux setting in my .tmux.conf is 

`set -g update-environment 'DISPLAY SSH_ASKPASS GPG_TTY SSH_AUTH_SOCK SSH_AGENT_PID SSH_CONNECTION WINDOWID XAUTHORITY TERM'`

I tried setting `if "[[ ${TERM} =~ 256color ]]" 'set -g default-terminal screen-256color'` which gives me screen-256color inside tmux in a bash shell but it still did not resolve the issue.

Does someone have any clue?

**Comment by nicm** [comment]
This issue does not include enough information. Please read the [CONTRIBUTING](https://raw.githubusercontent.com/tmux/tmux/master/CONTRIBUTING) file.

**Comment by nicm** [comment]
Oh, sorry you did include the version.

I don't understand your explanation. What exactly does this:
~~~~
echo $TERM
~~~~
Show right before you start tmux, and right after?

What tmux showenv says doesn't matter, only what is actually in TERM before programs are started.

**Comment by nicm** [comment]
Sounds like one of your shell profiles is in fact changing TERM. Are you sure you don't change it in any profile at all?

**Comment by dgengtek** [comment]
Thank you for your support.

I double checked and I did not grep any TERM in any profile or bash init scripts.

Starting urxvt my TERM is set to **rxvt-unicode-256color**
Starting tmux inside urxvt my TERM is set to **screen-256color**

The drawing issues only occur when I start ranger with `$ tmux neww ranger`. If I run ranger from the shell directly in a tmux window `$ ranger` I have no issues at all.

My used versions for reference:

4.18.5-arch1-1-ARCH #1 SMP PREEMPT Fri Aug 24 12:48:58 UTC 2018 x86_64 GNU/Linux
rxvt-unicode (urxvt) v9.22 - released: 2016-01-23
tmux 2.7
ranger version: ranger-master 1.9.1
Python version: 3.7.0 (default, Jul 15 2018, 10:44:58) [GCC 8.1.1 20180531]
Locale: en_US.UTF-8

**Comment by dgengtek** [comment]
I guess this issue can be closed since I did

```
tmux -Ltest kill-server
tmux -vv -Ltest -f/dev/null new
```
I was still having the issue. When I moved my .bashrc temporarly and redid the above I was unable to reproduce the issue which means that it is an issue with my environment setup and not tmux.

Here are the logs from `tmux -vv -Ltest -f/dev/null new` 
[tmux-server-23343.log](https://github.com/tmux/tmux/files/2364203/tmux-server-23343.log). Would be grateful if you know without digging further which environment variable could be responsible.

**Comment by nicm** [comment]
What do you have in .bashrc?

**Comment by dgengtek** [comment]
I do not know how my issue was raised because of these check lines `hash fzf 2>&1 | logger -t bashrc -p user.info` when using the export option since bash should  export variables and newly created functions only...

```
if hash fzf 2>&1 | logger -t bashrc -p user.info; then
  if hash fd 2>&1 | logger -t bashrc -p user.info; then
    FZF_DEFAULT_COMMAND='fd --type f --hidden --follow --exclude ".git"'
  elif hash rg 2>&1 | logger -t bashrc -p user.info; then
    FZF_DEFAULT_COMMAND='rg --files --no-ignore --hidden --follow --glob "!.git/*"'
  fi
  FZF_DEFAULT_OPTS="--bind alt-p:accept"
fi
```
Putting the logger lines outside the set -a, set +a fixed my issue.
